src/js/stores/library.js

```diff
@@ -26,9 +26,110 @@ define(function (require, exports, module) {
 
     var Fluxxor = require("fluxxor"),
         Immutable = require("immutable"),
+        EventEmitter = require("eventEmitter"),
         _ = require("lodash");
 
-    var events = require("../events");
+    var events = require("../events"),
+        log = require("js/util/log");
+    
+    /**
+     * LibrarySyncStatus is a wrapper class of the AdobeLibraryCollection instance to provide an easy way
+     * for listening to its sync status. Read LibrarySyncStatus#addSyncListener for more details of the 
+     * sync status.
+     */
+    var LibrarySyncStatus = function (libraryCollection) {
+        this._libraryCollection = libraryCollection;
+        this._syncingLibraryIDs = new Set();
+        this._emitter = new EventEmitter(this);
+        
+        this._status = {
+            isSyncing: false,
+            libraryNumber: this.getFilteredLibraries().length
+        };
+        
+        this._libraryCollection.addSyncListener(this._onLibraryCollectionSync.bind(this));
+    };
+    
+    /**
+     * Listen to libraries' sync status. 
+     *
+     * @param {function(isSyncing, libraryNumberChanged)} listener
+     *        isSyncing - if true, one or more libraries are uploading or downloading.
+     *        libraryNumberChanged - if true, the number of the libraries has increased or decresaed. 
+     * 
+     */
+    LibrarySyncStatus.prototype.addSyncListener = function (listener) {
+        this._emitter.addListener("sync", listener);
+    };
+    
+    /**
+     * Handle AdobeLibraryCollection sync event.
+     *
+     * @private
+     */
+    LibrarySyncStatus.prototype._onLibraryCollectionSync = function () {
+        log.debug("[CC Lib] on collection change");
+        
+        var newLibraries = this._libraryCollection.getNewLibrarySyncProgress();
+        
+        newLibraries.forEach(function (data) {
+            log.debug("[CC Lib] new library", data.progress);
+            
+            if (data.progress !== 100) {
+                this._syncingLibraryIDs.add(data.id);
+            } else {
+                this._syncingLibraryIDs.delete(data.id);
+            }
+        }, this);
+        
+        this._libraryCollection.libraries.forEach(function (library) {
+            var isLibrarySyncing = this._syncingLibraryIDs.has(library.id);
+            
+            if (!library.syncState && isLibrarySyncing) {
+                log.debug("[CC Lib] done library sync [", library.name, "]");
```

Comment:
can you tie all these log.debugs to a flag? Otherwise we're gonna have an issue in the future to stop them :)